### ISSUE TITLE
bug: assigning value to `ArraySection_t` fails inside a `WHERE` clause

### ISSUE THREAD

**Issue by kmr-srbh** [lfortran/lfortran#4317]
### Minimal Reproducible Example

```fortran
program main
   implicit none
   real, parameter :: zero = 0.0
   real :: array(1, 4)

   array(1, 1) = 0.0
   array(1, 2) = 1.0
   array(1, 3) = 0.0
   array(1, 4) = 1.0

   where (array(1, :) == zero)
      array(1, :) = 2.0
   end where

   print *, array

end program main
```

### Output
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ lfortran ./examples/example.f90
ASR verify pass error: The variable in ArrayItem must be an array, not a scalar
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 360, in __libc_start_main_impl()
  File "./csu/../sysdeps/x86/libc-start.c", line 58, in __libc_start_call_main()
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 2481, in main()
    return main_app(argc, argv);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 2442, in main_app(int, char**)
    err = compile_to_object_file(arg_file, tmp_o, false,
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 954, in compile_to_object_file()
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/fortran_evaluator.cpp", line 363, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 9967, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    pass_manager.apply_passes(al, &asr, co.po, diagnostics);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/pass/pass_manager.h", line 315, in LCompilers::PassManager::apply_passes(Allocator&, LCompilers::ASR::TranslationUnit_t*, LCompilers::PassOptions&, LCompilers::diag::Diagnostics&)
    apply_passes(al, asr, _passes, pass_options, diagnostics);
LCompilersException: Verify failed in the pass: where
```

This is a high hitting bug and prevents integrating subroutine `translv`, which is contained in it's own file, into the program as a general subroutine.

**Comment by assem2002** [comment]
found in files:
- mkba_sweep.f90 (lines : 332, 336, 341, 347)

**Comment by kmr-srbh** [comment]
This bug may be fixed by a combination of #4320 and #4323. I am guessing this based on the transformed Fortran code:
```fortran
! Fortran code after applying the pass: intrinsic_function
program main
implicit none
integer(4) :: __1_t
integer(4) :: __1_v
integer(4) :: __2_t
real(4), dimension(:), pointer :: __libasr__created__var__0__array_section_pointer_
logical(4), dimension(:), allocatable :: __libasr__created__var__1__real_comp_op_res
real(4), dimension(1, 4) :: array
real(4), parameter :: zero = 0.00000000e+00
array(1, 1) = 0.00000000e+00
array(1, 2) = 1.00000000e+00
array(1, 3) = 0.00000000e+00
array(1, 4) = 1.00000000e+00
__libasr__created__var__0__array_section_pointer_ => array(1, lbound(array, 2):ubound(array, 2))
deallocate(__libasr__created__var__1__real_comp_op_res)
allocate(__libasr__created__var__1__real_comp_op_res((ubound(array, 2) - lbound(array, 2))/1 + 1))
__1_v = lbound(__libasr__created__var__0__array_section_pointer_, 1)
do __1_t = lbound(__libasr__created__var__1__real_comp_op_res, 1), ubound(__libasr__created__var__1__real_comp_op_res, 1)
    __libasr__created__var__1__real_comp_op_res(__1_t) = __libasr__created__var__0__array_section_pointer_(__1_v) == zero
    __1_v = __1_v + 1
end do
where (__libasr__created__var__1__real_comp_op_res)
    do __2_t = lbound(array, 2), ubound(array, 2), 1
        array(1, __2_t) = 2.00000000e+00
    end do
end where
print *, array
end program main
```

where `__libasr__created__var__1__real_comp_op_res` is declared as `logical(4), dimension(:), allocatable :: __libasr__created__var__1__real_comp_op_res`.
